### PR TITLE
Fix race condition in proc_lib:stop/3 with small timeouts

### PR DESCRIPTION
**Pull request by juhlig**
This PR is a fix for issue #4853 by making sure the terminate message is always sent to the target process. It also removes the proxy process which was (probably) only there to catch late replies. It keeps the current outward behavior in regards to the format of exit messages. The stop test was augmented to make sure that the process eventually exits (ie, that the termination message was sent) even with a timeout of zero.

### CODE REVIEW

**Comment by Maria-12648430** [comment]
Looking good 😃

**Review by garazdawi — COMMENTED** [review]
This looks good. Just have some minor styling comments. Putting it into testing to see what pops up.

File: lib/stdlib/test/proc_lib_SUITE.erl [hunk 1]
```diff
@@ -668,34 +668,46 @@ stop(_Config) ->
...
     {'EXIT',timeout} = (catch proc_lib:stop(Pid3,normal,1000)),
 
+    %% Ensure that a termination message is always sent to the
+    %% target process and that it eventually terminates.
+    Pid4 = proc_lib:spawn(HangProc),
+    Ref4 = monitor(process, Pid4),
+    {'EXIT', timeout} = (catch proc_lib:stop(Pid4, normal, 0)),
+    ok = receive
+	{'DOWN', Ref4, process, _, _} ->
+	    ok
```
[the remark this answers is not in the document]
Comment (juhlig) [reply lib/stdlib/test/proc_lib_SUITE.erl:678]: Hm, why?

With the proxy process gone, this is all executed in the caller process. So, what if there is a message pending in the caller process' inbox?

[the remark this answers is not in the document]
Comment (juhlig) [reply lib/stdlib/test/proc_lib_SUITE.erl:678]: Oh, wait, this is the test suite XD

**Comment by garazdawi** [comment]
The tests seem to pass so I'm getting ready to merge this. Would you mind squashing all the commits and expanding the commit message with a larger description? the one in this PR should be enough.

**Comment by juhlig** [comment]
@garazdawi I squashed and elaborated the commit message as requested.

**Comment by garazdawi** [comment]
Thanks!

**Pull request merged**